backend/icpy/api/endpoints/media.py

```
@@ -50,15 +52,74 @@ async def get_image(
+                logger.debug(f"[Media API] Cache check failed for {image_id}: {cache_err}")
+
+        # If thumbnail requested and base64 is available, return it directly (fast path)
+        if thumbnail and image_ref.thumbnail_base64:
+            try:
+                content = base64.b64decode(image_ref.thumbnail_base64)
+                return Response(
+                    content=content,
+                    media_type="image/webp",
+                    headers={"Content-Disposition": f"inline; filename=thumb_{image_ref.current_filename}.webp"}
+                )
+            except Exception as thumb_b64_err:
```

Ruff (0.14.0)

Do not catch blind exception: `Exception`

(BLE001)